Autopilot/Src/timeStampingPOGI.cpp

```diff
@@ -0,0 +1,64 @@
+#include "timeStampingPOGI.hpp"
+#include "tim.h"
+#include "stm32f7xx_hal_tim.h"
+
+//Allocates and initializes the static data member pointer (not the object itself)
+TimeStampingPOGI* TimeStampingPOGI::s_Instance = nullptr;
+
+TimeStampingPOGI::TimeStampingPOGI(){
+
+    //initializing and starting timer
+    HAL_TIM_Base_Init(&htim14);
+    HAL_TIM_Base_Start(&htim14);
+
+}
+
+
+//just typical singleton stuff
+//creates the instance if it doesnt exist & returns the pointer to the instance
+TimeStampingPOGI* TimeStampingPOGI::GetInstance(){
+
+    if(s_Instance == nullptr){
+        s_Instance = new TimeStampingPOGI();
+    }
+
+    return s_Instance;
+}
+
+
+void TimeStampingPOGI::setGPSTime(GpsData_t* gpsData){
+
+    //check if its new and set the recentUTCTime to it
+    if (gpsData->timeIsNew){
+
+        __HAL_TIM_SET_COUNTER(&htim12, 0);
```

Comment:
htim14

If you are changing a variable name, a good practice is to always check with CTRL-F if you have changed all instances ;))